doc/concepts/morphology.rst

```diff
@@ -3,56 +3,117 @@
+as a set of round electrical cables, which have, among others properties,
+capacitances and resistances. These can in turn be expressed as a system of
+partial differential equations which can be solved to compute a current or
+voltage. The geometry of the biological cell is approximated as a tree of
+cable segments, which is a :term:`morphology` in Arbor nomenclature.
 
-The building blocks of a morphology tree are points and segments.
-A *point* is a three-dimensional location and a radius, used to mark the cross-sectional
-centre and radius of the cable.
+Let's first define the building blocks out of which we can construct a
+proper definition for a morphology:
 
-.. csv-table::
+.. glossary::
+
+  mpoint
+    A point in 3D space has three coordinates. In Arbor, we add a fourth coordinate: radius.
+
+.. csv-table:: The properties of a :term:`point <mpoint>`, in the context of cable cell morphologies.
    :widths: 10, 10, 30
+   :align: center
 
    **Field**,   **Type**, **Description**
-   ``x``,       real, x coordinate of centre of cable (μm).
-   ``y``,       real, y coordinate of centre of cable (μm).
-   ``z``,       real, z coordinate of centre of cable (μm).
+   ``x``,       real, x coordinate of center of cable (μm).
+   ``y``,       real, y coordinate of center of cable (μm).
+   ``z``,       real, z coordinate of center of cable (μm).
    ``radius``,  real, cross sectional radius of cable (μm).
 
+.. glossary::
 
-A *segment* is a frustum (cylinder or truncated cone), with the centre and radius at each
-end defined by a pair of points. In other words, in Arbor the radius between two points is interpolated
-linearly, resulting in either a cylinder (equal radii) or truncated cone (differing radii),
-centred at the line through the pair of points.
+  segment
+    A segment is a frustum (cylinder or truncated cone), with the center and radius at each
+    end defined by a pair of :term:`points <mpoint>`. In other words, in Arbor the radius between two points is interpolated
+    linearly, resulting in either a cylinder (equal radii) or truncated cone (differing radii),
+    centred at the line through the pair of points.
 
-.. csv-table::
+.. csv-table:: The properties of a :term:`segment`.
    :widths: 10, 10, 30
+   :align: center
 
-   **Field**,   **Type**, **Description**
+   **Field**,      **Type**,                           **Description**
    ``prox``,       :py:class:`point <arbor.mpoint>`,   the center and radius of the proximal end.
    ``dist``,       :py:class:`point <arbor.mpoint>`,   the center and radius of the distal end.
-   ``tag``,        integer, tag meta-data.
+   ``tag``,        integer (:term:`tag`),              "differentiate soma, axon, dendrite, etc."
+
+.. figure:: ../gen-images/term_segments.svg
+  :width: 300
+  :align: center
+
+  Four segments arranged head to toe.
+
+.. glossary::
+
+  branch
+    A branch is an unbranched sequence of :term:`segments <segment>`.
```

Comment:
It's the longest possible sequence of unbranched `segments`. Also the image does not accurately describe what a branch is, because it implies that any group of unbranched segments can be a branch.

Reply:
1. Can't they?
2. I'll make it "A branch is the longest possible unbranched sequence of segments" unless you are telling me branched segments exist!

Reply:
1. Nope they cant.
2. I worded that incorrectly ^^. I meant what you wrote.

Reply:
1. Then what do you propose, because to me this seems quite clear and correct. Maybe you mean that the segments should be joined?
2. OK!

Reply:
I mis-worded a lot of my first comment, let me try again: 

> A branch is the longest possible unbranched sequence of segments. Also the image does not accurately describe what a branch is, because it implies that any unbranched group of segments can be a branch.

What I mean to say is that the wording and image do not accurately represent the "longest possible" part of the definition. It implies that any group of segments can be a branch as long as there is no fork in the midst. But actually, a branch needs to have a forking point (or root/terminal) at either end to be a branch. Because the image is not part of a bigger branching structure, it doesn't really illustrate that concept.

Reply:
OK, if I understand you correctly, the absence of branching points makes it unclear. I thought of that, but then I could no longer have a sequence of segment>branch>cable with the same geometry without first discussing yet another concept (branching point). This sequence I want to keep. And strictly speaking, there is nothing wrong: this is the correct branch for that set of segments.

If the reader makes it to the morphology section, the morph-label-morph-fig shows what you mean. I think that is enough: we can't explain/show every detail at once.

Reply:
I have no problem with not modifying the image, as long as the wording is clear.

Reply:
The figure caption does not make any claims other than that it is the branch which corresponds to the previously shown segments, which is correct. I fail to see what's not clear.

Reply:
Looks fine!I hadn't checked the changes.